include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -4183,10 +4317,22 @@ MGTransferMF<dim, Number>::build(
   const std::vector<std::shared_ptr<const Utilities::MPI::Partitioner>>
     &external_partitioners)
 {
-  this->initialize_internal_transfer(dof_handler, this->mg_constrained_dofs);
-  this->initialize_transfer_references(internal_transfer);
+  const bool use_local_smoothing =
+    this->transfer.n_levels() == 0 || this->internal_transfer.n_levels() > 0;
+
+  if (use_local_smoothing)
+    {
+      this->initialize_internal_transfer(dof_handler,
+                                         this->mg_constrained_dofs);
+      this->initialize_transfer_references(internal_transfer);
+    }
+
   this->build(external_partitioners);
-  this->fill_and_communicate_copy_indices(dof_handler);
+
+  if (use_local_smoothing)
+    this->fill_and_communicate_copy_indices(dof_handler);
+  else
+    this->fill_and_communicate_copy_indices_global_coarsening(dof_handler);
```

Comment:
Can we run this only conditionally and skip the global coarsening part in case the `dof_handler` is the same as in the two-level transfer? Or maybe it already is because we call another function. (It is hard to see exactly where we end up without looking at a trace of common cases.)